src/cloudai/cli/handlers.py

```diff
@@ -233,6 +247,29 @@ def handle_dry_run_and_run(args: argparse.Namespace) -> int:
     if args.mode == "run" and not result.success:
         logging.error("CloudAI has not been installed. Please run install mode first.")
         logging.error(result.message)
+        return False
+
+    return True
+
+
+def handle_dry_run_and_run(args: argparse.Namespace) -> int:
+    setup_result = _setup_system_and_scenario(args)
+    if setup_result == (None, None, None):
+        return 1
+
+    system, test_scenario, tests = setup_result
+    assert system is not None
```

Comment:
Since this is not a test, user might a have a confusing error is any of these object is not defined. Let's handle each parsing separately as I mentioned above and show actionable messages instead of asserts (likely, not here, but during parsing).